AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1114,26 +1117,61 @@ - (void)__layout
   ASDisplayNodeAssertMainThread();
   ASDN::MutexLocker l(_propertyLock);
   CGRect bounds = self.bounds;
+
+  [self measureNodeWithBoundsIfNecessary:bounds];
+
   if (CGRectEqualToRect(bounds, CGRectZero)) {
     // Performing layout on a zero-bounds view often results in frame calculations
     // with negative sizes after applying margins, which will cause
     // measureWithSizeRange: on subnodes to assert.
     return;
   }
-
+  
   // Handle placeholder layer creation in case the size of the node changed after the initial placeholder layer
   // was created
   if ([self _shouldHavePlaceholderLayer]) {
     [self _setupPlaceholderLayerIfNeeded];
   }
   _placeholderLayer.frame = bounds;
-
+  
   [self layout];
   [self layoutDidFinish];
 }
 
+- (void)measureNodeWithBoundsIfNecessary:(CGRect)bounds
+{
+  // Normally measure will be called before layout occurs. If this doesn't happen, nothing is going to call it at all.
+  // We simply call measureWithSizeRange: using a size range equal to whatever bounds were provided to that element
+  if (self.supernode == nil && !self.supportsRangeManagedInterfaceState && [self _hasDirtyLayout]) {
+    if (CGRectEqualToRect(bounds, CGRectZero)) {
+      LOG(@"Warning: No size given for node before node was trying to layout itself: %@. Please provide a frame for the node.", self);
+    } else {
+      [self measureWithSizeRange:ASSizeRangeMake(bounds.size, bounds.size)];
+    }
+  }
+}
+
+- (void)layout
+{
+  ASDisplayNodeAssertMainThread();
+  
+  if ([self _hasDirtyLayout]) {
```

Comment:
I just replaced this method, before the `_hasDirtyLayout` change it was `!_flags.isMeasured` so the logic did not change. I have to talk with @nguyenhuy about this but this could be the same reason as in here: #1902